theme/management/commands/report_resource_size_vs_quota_inconsistency.py

```diff
@@ -0,0 +1,162 @@
+                    res_size = res.size
+                    print(f"Resource {res.short_id} is held by {user.username}, size: {res_size} bytes")
+                    total_size += res_size
+            converted_total_size_django = convert_file_size_to_unit(int(total_size), 'gb')
+            print(f"Total size of resources in Django: {converted_total_size_django} GB")
+
+            if not math.isclose(used_value_irods_dz, converted_total_size_django, abs_tol=0.1):
+                # report inconsistency
+                report_dict = {
+                    'user': uq.user.username,
+                    'django': converted_total_size_django,
+                    'irods': used_value_irods_dz}
+                quota_report_list.append(report_dict)
+                print('quota incosistency: {} reported in django vs {} reported in iRODS for user {}'.format(
+                    converted_total_size_django, used_value_irods_dz, user.username), flush=True)
+
+                if update:
+                    print("Attempting to fix the inconsistency by updating file sizes in django")
+                    res_files = []
+                    for res in held_resources:
+                        print(f"Total files in resource {res.short_id}: {res.files.all().count()}")
+                        print(f'{current_site}/resource/{res.short_id}: currently {res.size} bytes')
+                        file_counter = 0
+                        # exclude files with size 0 as they don't exist in iRODS
+                        for res_file in res.files.exclude(_size=0).iterator():
+                            # this is an expensive operation (3 irods calls per file) - about 1 min for 100 files
+                            # size, checksum and modified time are obtained from irods and assigned to
+                            # relevant fields of the resource file object
+                            res_file.set_system_metadata(resource=res, save=False)
+                            res_files.append(res_file)
+                            file_counter += 1
+                            print(f"Updated file count: {file_counter}")
+                            if res_file._size <= 0:
+                                print(f"File {res_file.short_path} was not found in iRODS.")
+
+                        if res_files:
+                            ResourceFile.objects.bulk_update(res_files,
+                                                             ResourceFile.system_meta_fields(), batch_size=_BATCH_SIZE)
+                            print(f"Updated {file_counter} files for resource {res.short_id}")
+                            res.refresh_from_db()
+                            print(f'{current_site}/resource/{res.short_id}: now {res.size} bytes')
+                        else:
+                            print(f"Resource {res.short_id} contains no files.")
+
+                elif reset:
+                    print("Resetting file size cache in django")
```

Comment:
I am wondering why would we want to reset file sizes. If a resource with 100s of files get reset, we may get timeout error in accessing the landing page for that resource.

Reply:
That's true. My thought was that it might be useful to be able to do this in a case by case basis, when we are aware there is a problem with a user quota but we don't want to take the time to run the update at the time this script is run. If you think best to remove that, I will do so...

Reply:
I am just trying to understand what problem will be solved if we reset the file size. So if we found an issue with user quota, and we reset the file size, the file size will be recomputed next time only when the resource is accessed by the user. And if there are multiple resources for which files were reset, then unless each of these resources are accessed, we will still have the quota issue? May be the nightly  celery task would update the file size for all these resources?

Reply:
It is a good point that resetting would effectively "kick the can" so that the computation had to either be done at the time of resource access or during the nightly file metadata cache update task. But maybe you're right that this isn't the best way to handle.

My thinking is: I am aware that our cached filesizes are incorrect for a large number of our resourcefiles (I think perhaps as much as 10% of the files in our system). If we kicked off the management command with `--update` it would iterate through very slowly and finish... in weeks?

So instead of that approach, I was thinking that we would only run with `--update` for large resources with identified filesize mismatch, on a one-off basis. These are resources where resource landing page might time-out during calculation.

For the other majority of resources with discrepancies, we could run `--reset` to be able to iterate through more quickly (without running the `iquest` against irods to get filesize which is the slow part of the `--update`) and rely on page load or nightly task to update the cache.

Reply:
Thanks for explaining the scenarios where one would use `--update` vs` --reset`. I am fine with leaving the `--reset` option. If we are relying on nightly task to update the file sizes, then you may want to check that the resource '`updated`' field gets updated as part of the `--reset `operation in the command. I assume we can update the '`updated`' field for any published resources.

Reply:
Thanks @pkdash! I added a `--min_size` param for the management command and also set `res.updated` in the case of `--reset`. Good suggestions, thanks.